### ISSUE TITLE
`@stylistic/selector-descendant-combinator-no-non-space` passes over a selector whose comment the parser moves across a parenthesised group

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#159]
Found while fixing [#124](https://github.com/stylelint-stylistic/stylelint-stylistic/issues/124), which taught the rule to read a selector carrying a comment. A comment standing in front of a parenthesised group is a corner the parser does not survive, and such a selector is passed over for now.

### What minimal example or steps are needed to reproduce the bug?

```css
.foo /*comment*/	( )	.bar {}
```

A tab, or a line break, or any whitespace that is not a single space has to follow the group; a single space there and the parser reads the selector correctly.

### What minimal configuration is needed to reproduce the bug?

```json
{
  "plugins": [
    "@stylistic/stylelint-plugin"
  ],
  "rules": {
    "@stylistic/selector-descendant-combinator-no-non-space": true
  }
}
```

### How did you run Stylelint?

```shell
stylelint "**/*.css" --config myconfig.json --fix
```

### What did you expect to happen?

The tab after the group is reported and collapsed, the comment staying in front of the group where the author put it.

### What actually happened?

Nothing is reported. The rule passes over any selector that `postcss-selector-parser` does not give back as it took it.

### Do you have a proposal to fix the bug?

A parenthesised group standing where a combinator belongs is nothing CSS has a name for, and nothing the parser has a place for either. It reads the group as a combinator whose `value` is ` ( )` and whose `sourceIndex` is the index of the space in front of the *comment*, and files the comment itself into the `raws.spaces.after` of that same node. So the node claims a stretch of the file it does not stand in, and stringifying the tree gives `.foo ( )/*comment*/\t\t.bar` — the comment has crossed the group, and a tab has appeared where none was written.

Three things follow, and all three are unanswerable while the tree says this:

- the text of the warning is a text the file does not hold, `" ( )/*comment*/\t"` for the example above;
- the position it is reported at points into the comment rather than at the group;
- a fix to anything else in the same selector writes the whole re-stringified selector back, carrying the reordering with it — and the result is not stable, since a second run reads the moved comment differently again.

The rule therefore compares what the parser gives back with what it was handed, and passes the selector over when the two differ. Answering this properly means either repairing the reading upstream in `postcss-selector-parser`, or reading such a group out of the selector text directly instead of out of the tree.

**Comment by firefoxic** [comment]
One thing the workaround costs, worth writing down before it is forgotten: the comparison is made against the whole selector of a rule node, so a list loses its other selectors along with the one the parser mangles. `.foo  .bar, .baz /*comment*/\t( )\t.qux {}` reports nothing, though the two spaces in the first selector have nothing to do with the group in the second.

Narrowing the reading to one selector of the list would be easy enough — each has a `sourceIndex` and a text of its own to compare. Narrowing the *writing* would not: a fix hands back the whole tree stringified, so a mangled selector anywhere in the list would ride along into the output. The list could be read selector by selector and reported on, with every fix refused as long as one of them is mangled; that is more machinery than the corner has been worth so far.
